unit_tests/test_scan_operation_thread.py

```diff
@@ -0,0 +1,210 @@
+"""
+Unit tests for scan_operation_thread.py
+contains 3 tests that check scan operations behavior in different conditions
+
+test_scan_positive - positive scenario
+test_scan_negative_operation_timed_out - getting operation_timed_out in scan execution
+test_scan_negative_exception - getting operation_timed_out in scan execution (with and without nemesis)
+"""
+from pathlib import Path
+import os
+import logging
+import sys
+from threading import Event
+from unittest.mock import MagicMock, patch
+import pytest
+from cassandra import OperationTimedOut
+from unit_tests.test_cluster import DummyDbCluster, DummyNode
+from sdcm.scan_operation_thread import ScanOperationThread, FullScanParams
+
+logger = logging.getLogger()
+logger.level = logging.DEBUG
+stream_handler = logging.StreamHandler(sys.stdout)
+logger.addHandler(stream_handler)
+
+
+DEFAULT_PARAMS = {
+    'termination_event': Event(),
+    'fullscan_user': 'sla_role_name',
+    'fullscan_user_password': 'sla_role_password',
+    'duration': 10,
+    'interval': 0,
+    'validate_data': True
+}
+
+
+class DBCluster(DummyDbCluster):  # pylint: disable=abstract-method
+    # pylint: disable=super-init-not-called
+    def __init__(self, connection_mock, nodes, params):
+        logger.debug(nodes)
+        logger.debug(params)
+        super().__init__(nodes, params=params)
+        self.connection_mock = connection_mock
+        self.params = {'nemesis_seed': 1}
+
+    def get_non_system_ks_cf_list(*args, **kwargs):
+        # pylint: disable=unused-argument
+        # pylint: disable=no-method-argument
+        return ["test", "a.b"]
+
+    def cql_connection_patient(self, *args, **kwargs):
+        # pylint: disable=unused-argument
+        return self.connection_mock
+
+
+def get_event_log_file(events):
+    if (log_file := Path(events.temp_dir, "events_log", "events.log")).exists():
+        return log_file.read_text(encoding="utf-8").rstrip().split('\n')
+    return ""
+
+
+@pytest.fixture(scope='function', autouse=True)
+def cleanup_fife(events):
```

Comment:
why do we need this cleanup ? 

and if we do, why not removing the files ? why clearing the content ?

and what's fife ?